File: SIN BDD/SOAP_JAVA_CONUNI_GR06/07. CLIMOVIL/cliente_movil_soap_java_conuni_gr06/app/src/main/java/ec/edu/monster/modelo/ServicioTemperatura.java
package ec.edu.monster.modelo;

import java.util.LinkedHashMap;
import java.util.Map;

public class ServicioTemperatura {

    private final ClienteSoap clienteSoap = new ClienteSoap();

    public double celsiusAFahrenheit(double celsius) throws Exception {
        return invocarUnario("celsiusAFahrenheit", "celsius", celsius);
    }

    public double fahrenheitACelsius(double fahrenheit) throws Exception {
        return invocarUnario("fahrenheitACelsius", "fahrenheit", fahrenheit);
    }

    public double celsiusAKelvin(double celsius) throws Exception {
        return invocarUnario("celsiusAKelvin", "celsius", celsius);
    }

    public double kelvinACelsius(double kelvin) throws Exception {
        return invocarUnario("kelvinACelsius", "kelvin", kelvin);
    }

    public double fahrenheitAKelvin(double fahrenheit) throws Exception {
        return invocarUnario("fahrenheitAKelvin", "fahrenheit", fahrenheit);
    }

    private double invocarUnario(String operacion, String nombreParametro, double valor) throws Exception {
        Map<String, String> parametros = new LinkedHashMap<>();
        parametros.put(nombreParametro, String.valueOf(valor));
        String respuesta = clienteSoap.invocar(operacion, parametros);
        return Double.parseDouble(respuesta);
    }
}

File: SIN BDD/SOAP_DOTNET_CONUNI_GR06/05. CLIESCRITORIO/cliente_escritorio_soap_dotnet_conuni_gr06/src/Modelo/ServicioTemperatura.cs
namespace Ec.Edu.Monster.Modelo;

public class ServicioTemperatura
{
    private readonly ClienteSoap cliente = new();

    public double CelsiusAFahrenheit(double celsius) => cliente.Ejecutar(servicio => servicio.CelsiusAFahrenheit(celsius));

    public double FahrenheitACelsius(double fahrenheit) => cliente.Ejecutar(servicio => servicio.FahrenheitACelsius(fahrenheit));

    public double CelsiusAKelvin(double celsius) => cliente.Ejecutar(servicio => servicio.CelsiusAKelvin(celsius));

    public double KelvinACelsius(double kelvin) => cliente.Ejecutar(servicio => servicio.KelvinACelsius(kelvin));

    public double FahrenheitAKelvin(double fahrenheit) => cliente.Ejecutar(servicio => servicio.FahrenheitAKelvin(fahrenheit));

    public double KelvinAFahrenheit(double kelvin) => cliente.Ejecutar(servicio => servicio.KelvinAFahrenheit(kelvin));
}
